### PR TITLE
chore: route WalletAction, WalletSpendAction to walletFactory contract

### PR DESCRIPTION
**Pull request by dckc**
refs: #4398 

## Description

route WalletAction, WalletSpendAction through the `bridgeManager` to walletFactory contract

 - cosmic-swingset integration test helpers:: fund-acct, wallet-action, provision-acct Makefile rules

### Security Considerations

relies on golang to authenticate the `.owner` address before we use it to look up the relevant wallet.

### Documentation Considerations

?

### Testing Considerations

Tested manually, so far.

Automated testing golang+JS stuff is beyond the current state-of-the-art.

### CODE REVIEW

File: packages/vats/src/bridge.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ export function makeBridgeManager(E, D, bridgeDevice) {
     // );
 
     // Notify the specific handler, if there was one.
-    E(srcHandlers.get(srcID)).fromBridge(srcID, obj);
+    void E(srcHandlers.get(srcID)).fromBridge(srcID, obj);
```
Comment (turadg) [inline packages/vats/src/bridge.js:49]: 👍

File: packages/wallet/contract/src/walletFactory.js [hunk 2]
```diff
@@ -10,22 +10,32 @@ import '@agoric/zoe/exported.js';
...
 
 /**
  * @typedef {{
  *   agoricNames: NameHub,
  *   board: Board,
  *   namesByAddress: NameHub,
  * }} SmartWalletContractTerms
+ *
+ * @typedef {{
+ * 	 type: string, // WALLET_ACTION
```
Comment (turadg) [inline packages/wallet/contract/src/walletFactory.js]: is the type `ActionType`? 
```suggestion
 * type: keyof ActionType
```

File: packages/wallet/contract/src/walletFactory.js [hunk 3]
```diff
@@ -34,12 +44,32 @@ export const start = async (zcf, privateArgs) => {
...
+     * @param {WalletAction} obj
+     */
+    fromBridge: async (srcID, obj) => {
+      console.log('walletFactory.fromBridge:', srcID, obj);
+      assert(obj, 'missing wallet action');
+      assert.typeof(obj, 'object');
+      assert.typeof(obj.owner, 'string');
+      const wallet = walletsByAddress.get(obj.owner); // or throw
+      console.log('walletFactory:', { wallet });
+      return E(wallet).performAction(obj); // TODO: add performAction to lib-wallet.js
```
Comment (turadg) [inline packages/wallet/contract/src/walletFactory.js]: I'm not sure what value `fromBridge` has when `performAction` doesn't exist but I'll trust you do :)

**Review by turadg — APPROVED** [review]
No blocking concerns.

File: packages/wallet/contract/src/walletFactory.js [hunk 4]
```diff
@@ -34,12 +44,32 @@ export const start = async (zcf, privateArgs) => {
...
   const walletsByAddress = makeScalarBigMapStore('walletsByAddress');
   const provider = makeAtomicProvider(walletsByAddress);
 
+  const handleWalletAction = Far('walletActionHandler', {
+    /**
+     *
+     * @param {string} srcID
+     * @param {WalletAction} obj
+     */
+    fromBridge: async (srcID, obj) => {
```
Comment (samsiegart) [inline packages/wallet/contract/src/walletFactory.js:66]: Just thinking out loud, we will at some point want `provideSmartWallet` to be callable in this same way so a user can create their wallet from the browser.

**Review by samsiegart — APPROVED** [review]

**Comment by dckc** [comment]
converting back to draft... I think this only does `WALLET_ACTION` and not `WALLET_SPEND_ACTION`.

**Comment by dckc** [comment]
another reason to convert back to draft: we have plans to migrate from `WalletAction` to something else: https://github.com/Agoric/agoric-sdk/issues/4398#issuecomment-1200531537

File: packages/wallet/contract/src/walletFactory.js (hunk 2 above)
Comment (dckc) [reply packages/wallet/contract/src/walletFactory.js]: yes, but `ActionType` is written as a module namespace, not a record, so `keyof` doesn't work.

I have a refinement pending.

File: packages/wallet/contract/src/walletFactory.js (hunk 3 above)
Comment (dckc) [reply packages/wallet/contract/src/walletFactory.js]: This didn't get past `yarn lint` so made the whole like a TODO comment.

File: packages/wallet/contract/src/walletFactory.js (hunk 4 above)
Comment (dckc) [reply packages/wallet/contract/src/walletFactory.js:66]: For follow-up, see notes in #4654

**Pull request merged**